server/src/main/java/org/elasticsearch/index/mapper/ConstantFieldType.java

```diff
@@ -88,7 +88,7 @@ public final Query termQueryCaseInsensitive(Object value, QueryShardContext cont
     }
 
     @Override
-    public final Query termsQuery(List<?> values, QueryShardContext context) {
+    public final Query termsQuery(Collection<?> values, QueryShardContext context) {
```

Comment:
Here are some notes about this modification
1. Using `Iterable` to load data from `BytesRef` lazily is very good, but the `MappedFieldType#termsQuery` used to expect a `list` of terms, which means we still had to convert the `Iterable` to `List` to build a TermsQuery. This can make the lazily load meaningless, so i changed the method signature from a `List` to a `Collection`.
2. The reason that I did not convert `List` to `Iterable` but `Collection` is that we usually use a `List#size()` to pre-construct another Collection, e.g. a `LongHashSet` in `DateScriptFieldType`. This can help when we add data into the new collection. we can not get a `size` value from `Iterable`, but we can get it from `Collection`.

Reply:
++ to move from a List to a Collection in `termsQuery`. That touches a lot of files but the change makes sense to me.